src/app/regions-filter/regions-filter.component.ts

```diff
@@ -1,28 +1,32 @@
 import { RegionsFilter } from './regions-filter';
-import { Component, ElementRef, OnInit, ViewChild } from '@angular/core';
+import { Component, ElementRef, Input, OnInit, ViewChild } from '@angular/core';
 import { Store } from '@ngxs/store';
-import { RegionsFilterState, SetRegionsFilter } from './regions-filter.state';
+import { RegionsFilterModel, RegionsFilterState, SetRegionsFilter } from './regions-filter.state';
 import { ValidateNested } from 'class-validator';
 import { StatefulComponent } from 'app/common/stateful-component';
+import { DatasetsService } from 'app/datasets/datasets.service';
 
 @Component({
   selector: 'gpf-regions-filter',
   templateUrl: './regions-filter.component.html',
 })
 export class RegionsFilterComponent extends StatefulComponent implements OnInit {
-  @ValidateNested() public regionsFilter = new RegionsFilter(this.store);
+  @Input() public genome = '';
+  @ValidateNested() public regionsFilter = new RegionsFilter();
   @ViewChild('textArea') private textArea: ElementRef;
 
-  public constructor(protected store: Store) {
+  public constructor(protected store: Store, private datsetsService: DatasetsService) {
```

Comment:
typo: `datsetsService`
also, why is the datasets service injected here? I don't see it being used